### PR TITLE
trigger custom click handler upon marker click

### PR DESCRIPTION
**Pull request by ereztison**
It would be useful to have a custom click handler fired when the user clicks a marker, so developers will be able to run their own code when that happens.

To use this implementation simply add a "onMarkerClicked" field (function type) to a marker JSON, for example:
```
player.markers.add([{ 
  time: 1, 
  text: "ABC", 
  onMarkerClicked: function() { console.log("marker clicked"); }
}]);
```

### CODE REVIEW

File: src/videojs.markers.js [hunk 1]
```diff
@@ -96,6 +96,9 @@
             marker.div.on('click', function(e) {
                var key = $(this).data('marker-index');
                player.currentTime(markers[key].time);
+
+               if (marker.onMarkerClicked)
+                  marker.onMarkerClicked();
```
Comment (spchuang) [inline src/videojs.markers.js]: I would add full brackets {} .

File: dist/videojs-markers.js [hunk 2]
```diff
@@ -98,6 +98,9 @@
             marker.div.on('click', function(e) {
                var key = $(this).data('marker-index');
                player.currentTime(markers[key].time);
+
+               if (marker.onMarkerClicked)
+                  marker.onMarkerClicked();
```
Comment (spchuang) [inline dist/videojs-markers.js]: You don't need to directly edit dist files. They should be generated from "grunt compile" (see grunt file for detail)

**Comment by spchuang** [comment]
I would also rename the callback to simply "onClicked".

The function onMarkerClicked could take "marker" as the parameter to have access to marker time and text. 

In regards to @tjenkinson's comment on preventing default click behavior, I think using the return value of onMarkerClicked for that intent is confusing. I would suggest adding a new property for marker. Something like "preventDefaultClickGoto," (if you have a better naming, please use it instead!)

**Comment by ereztison** [comment]
Thanks @spchuang for the feedback, I updated the pull request with your suggestions.

Regarding preventing default click behaviour I tend to agree with @tjenkinson, returning false seems like [a jQuery practice] (http://stackoverflow.com/questions/1357118/event-preventdefault-vs-return-false) so I guess many developers are familiar with it, it also allows for determining if to cancel the event or not dynamically during the callback execution in a more straight forward manner.

There is also the event.preventDefault() javascript practice which is similar to what you suggested but with the difference it is invoked on the event rather than the marker.

My personal flavour is the "return false" though, but that's only my suggestion :)

**Comment by spchuang** [comment]
@ereztison, okay let's go for the "return false" approach.

Thanks for the update!

**Pull request merged**